browser/ui/webui/settings/brave_settings_clear_rewards_data_handler.cc

```diff
@@ -0,0 +1,234 @@
+#include "chrome/browser/browsing_data/chrome_browsing_data_remover_delegate_factory.h"  // NOLINT
+#include "chrome/browser/browsing_data/counters/browsing_data_counter_factory.h"
+#include "components/prefs/pref_member.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/browsing_data_filter_builder.h"
+#include "content/public/browser/web_ui.h"
+
+namespace {
+
+settings::RewardsDataType GetDataTypeFromDeletionPreference(
+    const std::string& pref_name) {
+  using DataTypeMap = base::flat_map<std::string, settings::RewardsDataType>;
+  static base::NoDestructor<DataTypeMap> preference_to_datatype(
+      std::initializer_list<DataTypeMap::value_type>{
+          {brave_rewards::prefs::kRewardsAutoContributeSites,
+              settings::RewardsDataType::REWARDS_AUTO_CONTRIBUTE},
+          {brave_rewards::prefs::kRewardsAllData,
+              settings::RewardsDataType::REWARDS_ALL_DATA},
+      },
+      base::KEEP_FIRST_OF_DUPES);
+
+  auto iter = preference_to_datatype->find(pref_name);
+  DCHECK(iter != preference_to_datatype->end());
+  return iter->second;
+}
+
+const char* kCounterPrefs[] = {
+  brave_rewards::prefs::kRewardsAutoContributeSites
+};
+
+} // namespace
+
+namespace settings {
+
+// ClearRewardsDataHandler ----------------------------------------------------
+
+ClearRewardsDataHandler::ClearRewardsDataHandler(content::WebUI* webui)
+    : profile_(Profile::FromWebUI(webui)),
+      weak_ptr_factory_(this) {
+  Profile* profile = Profile::FromWebUI(webui);
+  rewards_service_ =
+      brave_rewards::RewardsServiceFactory::GetForProfile(profile);
+  if (rewards_service_) {
+    rewards_service_->AddObserver(this);
+  }
+}
+
+ClearRewardsDataHandler::~ClearRewardsDataHandler() {
+  if (rewards_service_) {
+    rewards_service_->RemoveObserver(this);
+  }
+}
+
+void ClearRewardsDataHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      "clearRewardsData",
+      base::BindRepeating(&ClearRewardsDataHandler::HandleClearRewardsData,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "initializeClearRewardsData",
+      base::BindRepeating(&ClearRewardsDataHandler::HandleInitialize,
+                          base::Unretained(this)));
+
+  web_ui()->RegisterMessageCallback(
+      "isContributionInProgress",
+      base::BindRepeating(
+                &ClearRewardsDataHandler::HandleIsContributionInProgress,
+                          base::Unretained(this)));
+}
+
+void ClearRewardsDataHandler::OnJavascriptAllowed() {
+  DCHECK(counters_.empty());
+  for (const std::string& pref : kCounterPrefs) {
+    if (pref == brave_rewards::prefs::kRewardsAutoContributeSites) {
+      AddCounter(std::make_unique<RewardsCounter>(profile_),
+        browsing_data::ClearBrowsingDataTab::ADVANCED);
+    }
+  }
+}
+
+void ClearRewardsDataHandler::OnJavascriptDisallowed() {
+  weak_ptr_factory_.InvalidateWeakPtrs();
```

Comment:
why are you calling this?

Reply:
See `ClearBrowsingDataHandler::OnJavascriptDisallowed`